libcaf_net/caf/policy/doorman.hpp

```diff
@@ -0,0 +1,100 @@
+/******************************************************************************
+ *                       ____    _    _____                                   *
+ *                      / ___|  / \  |  ___|    C++                           *
+ *                     | |     / _ \ | |_       Actor                         *
+ *                     | |___ / ___ \|  _|      Framework                     *
+ *                      \____/_/   \_|_|                                      *
+ *                                                                            *
+ * Copyright 2011-2019 Dominik Charousset                                     *
+ *                                                                            *
+ * Distributed under the terms and conditions of the BSD 3-Clause License or  *
+ * (at your option) under the terms and conditions of the Boost Software      *
+ * License 1.0. See accompanying files LICENSE and LICENSE_ALTERNATIVE.       *
+ *                                                                            *
+ * If you did not receive a copy of the license files, see                    *
+ * http://opensource.org/licenses/BSD-3-Clause and                            *
+ * http://www.boost.org/LICENSE_1_0.txt.                                      *
+ ******************************************************************************/
+
+#pragma once
+
+#include <vector>
+
+#include "caf/logger.hpp"
+#include "caf/net/socket.hpp"
+#include "caf/net/stream_socket.hpp"
+#include "caf/net/tcp_accept_socket.hpp"
+#include "caf/net/tcp_stream_socket.hpp"
+#include "caf/policy/scribe.hpp"
+#include "caf/send.hpp"
+
+namespace caf {
+namespace policy {
+
+/// A doorman accepts TCP connections and creates scribes to handle them.
+class doorman {
+public:
+  doorman(net::tcp_accept_socket acceptor) : acceptor_(acceptor) {
+    // nop
+  }
+
+  net::tcp_accept_socket acceptor_;
+
+  net::socket handle() {
+    return acceptor_;
+  }
+
+  template <class Parent>
+  error init(Parent& parent) {
+    if (auto err = parent.application().init(parent))
+      return std::move(err);
+    parent.mask_add(net::operation::read);
+    return none;
+  }
+
+  template <class Parent>
+  bool handle_read_event(Parent& parent) {
+    auto sck = net::accept(acceptor_);
```

Comment:
I think we should not introduce new abbreviations:

```sh
fgrep -R 'sck' libcaf_net | wc -l
       0
```

```suggestion
    auto sock = net::accept(acceptor_);
```

Also ok IMO:

```c++
    auto x = net::accept(acceptor_);
```